core/src/staked_nodes_updater_service.rs

```diff
@@ -33,13 +34,21 @@ impl StakedNodesUpdaterService {
                 let mut last_stakes = Instant::now();
                 while !exit.load(Ordering::Relaxed) {
                     let mut stake_map = Arc::new(HashMap::new());
-                    if let Ok(true) = Self::try_refresh_ip_to_stake(
+                    match Self::try_refresh_pk_to_stake(
                         &mut last_stakes,
                         &mut stake_map,
                         &rpc_load_balancer,
                     ) {
-                        let shared = StakedNodes::new(stake_map, staked_nodes_overrides.clone());
-                        *shared_staked_nodes.write().unwrap() = shared;
+                        Ok(true) => {
+                            let shared =
+                                StakedNodes::new(stake_map, staked_nodes_overrides.clone());
+                            *shared_staked_nodes.write().unwrap() = shared;
+                        }
+                        Err(err) => {
+                            warn!("Failed to refresh pk to stake map! Error: {:?}", err);
+                            sleep(PK_TO_STAKE_REFRESH_DURATION);
```

Comment:
why do you need this sleep here?

Reply:
We don't want the thread to spam the RPC with a request that's consistently not working. 

Also makes sense to free up the CPU. 

`PK_TO_STAKE_REFRESH_DURATION` is only 5s so on the offchance a single request times out, it shouldn't really affect relayer operation